playbooks/roles/local.defaults/tasks/main.yml

```diff
@@ -1,47 +1,89 @@
+  register: config_yml
 
-- name: Load local settings
-  include_vars:
-    name: local_settings
-    file: "{{ lookup('env', 'PWD') | default('.') }}/local.yml"
-  failed_when: false
-
-- name: Combine settings
-  set_fact:
-    settings: >-
-      {{
-         global_settings |
-         combine(personal_settings, local_settings,
-                 recursive = true, list_merge = 'append_rp')
-      }}
+- name: Generate configuration
+  when: not config_yml.stat.exists
+  block:
+    - name: Load global settings
+      include_vars:
+        name: global_settings
+        file: settings.yml
 
-- debug:
-    var: settings
-
-- name: Generate config.yml in ansible directory
-  template:
-    src: config.yml.j2
-    dest: ./ansible/config.yml
-  vars:
-    os: "{{ use_distro | default('centos9') }}"
-    be: "{{ (backend | default('xfs') | split('.'))[0] }}"
-    variant: "{{ (backend | default('xfs') | split('.'))[1] | default('default') }}"
-    max_memory: "{{ ansible_memfree_mb }}"
-    max_cpu: "{{ ansible_processor_nproc }}"
+    - name: Load personal settings
+      include_vars:
+        name: personal_settings
+        file: ~/.site/settings.yml
+      failed_when: false
+
+    - name: Load local settings
+      include_vars:
+        name: local_settings
+        file: "{{ lookup('env', 'PWD') | default('.') }}/local.yml"
+      failed_when: false
+
+    - name: Combine settings
+      set_fact:
+        settings: >-
+          {{
+             global_settings |
+             combine(personal_settings, local_settings,
+                     recursive = true, list_merge = 'append_rp')
+          }}
+
+    - debug:
+        var: settings
+
+    - name: Generate config.yml in ansible directory
+      template:
+        src: config.yml.j2
+        dest: ./ansible/config.yml
+      vars:
+        os: "{{ use_distro | default('centos9') }}"
+        be: "{{ (backend | default('xfs') | split('.'))[0] }}"
+        variant: "{{ (backend | default('xfs') | split('.'))[1] | default('default') }}"
+        max_memory: "{{ ansible_memfree_mb }}"
+        max_cpu: "{{ ansible_processor_nproc }}"
 
 - name: Load configuration
   include_vars:
     file: "ansible/config.yml"
 
 - debug:
     var: config
+
+# TODO: Use openssh_keypair action
+- name: Create SSH key
+  command:
+    cmd: ssh-keygen -C "SITE SSH Key" -f ~/.site/id_rsa -N "" -t rsa -b 2048
+    creates: ~/.site/id_rsa
+
+- name: Update the inventory
+  add_host:
+    name: "{{ item }}"
+    ansible_user: root
+    ansible_port: 22
+    ansible_ssh_common_args: "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+    ansible_ssh_private_key_file: "~/.site/id_rsa"
+    groups: "{{ config.nodes[item].groups + ['site'] }}"
+    site: "{{ config.nodes[item] }}"
+  with_items: "{{ config.nodes.keys() }}"
+
+- name: Create state directory structure on host
+  file:
+    path: "{{ config.statedir }}/{{ item }}"
+    state: directory
+  loop: "{{ groups['site'] }}"
```

Comment:
The _local.defaults_ role is processed in the localhost, before creating any VM, so it's not executed in the _admin_ node. This task was previously done in the _provisioner.vagrant_ role, which also runs on the localhost before creating the VMs. Am I missing something ?

Reply:
Right, I think I mixed up reviewing various roles at the same time.

Reply:
Sorry to dig this up again. I think I know what I wanted to ask. Previously directory structure did not have one for `admin` node/s. But with this change directories are created for those nodes too.

Reply:
Oh, now I understand what you meant. I think we'll need to have a directory for all nodes (including `admin` ones) because in some cases (like the GPFS backend) even the `admin` nodes become active cluster members.

Reply:
In that case we'll have to expand the _nodes.statedump_ playbook to include `admin` group also in the list of hosts when GPFS backend is added. This would also mean that we add _logs_ and/or _config_ sub directories inside _playbooks/roles/sit.gpfs/_ to copy required logs and/or configuration during statedump process.